tools/old_config_maker/graph_node.py

```diff
@@ -0,0 +1,260 @@
+
+            master.destroy()
+            master.quit()
+
+        master = Tk()
+        master.title("Add Metric")
+        master.geometry('250x130')
+
+        Label(master, text="Label").grid(row=0)
+        Label(master, text="Function").grid(row=1)
+        Label(master, text="x_stream").grid(row=2)
+        Label(master, text="y_stream").grid(row=3)
+        Label(master, text="z_stream").grid(row=4)
+
+        entries = [Entry(master) for _ in range(5)]
+
+        for i, entry in enumerate(entries):
+            entry.grid(row=i, column=1)
+
+        Button(master, text="Add Metric", command=lambda: close_config_window(self, master, entries)).grid(row=9, column=0)
+
+        master.mainloop()
+
+    def delete(self):
+        """
+        Deletes all tkinter items in node.
+
+        Returns
+        -------
+        Self object so it can be deleted with all of its stuff in one line
+        """
+
+        for value in self.items.values():
+            if type(value) is dict:
+                for item in value.values():
+                    item.destroy()
+            else:
+                value.destroy()
+
+        return self
+
+    def set_grid(self, row_offset=None):
+        """
+        Every button/item has a grid based location and this places every item to that grid space
+
+        Parameters
+        ----------
+        row_offset: number, default=None
+            How many rows this graph_node should start down from top of frame
+        """
+
+        if row_offset: self.row_offset = row_offset
+
+        rolling_offset = self.row_offset  # If checkboxes take extra lines, the lines underneath will drop one
+
+        for key, value in self.items.items():
+            if key in self.item_locations.keys():
+                grid_values = self.item_locations[key]
+
+                if type(value) is dict:
+                    for i, key in enumerate(value.keys()):
+                        value[key].grid(sticky="W", column=i % GraphNode.checkbox_width,
+                                        row=rolling_offset + grid_values + int(i / GraphNode.checkbox_width))
+
+                    rolling_offset += int(len(value) / GraphNode.checkbox_width)
+                else:
+                    value.grid(column=grid_values[0],
+                               row=grid_values[1] + rolling_offset if grid_values[1] > 1 else self.row_offset,
+                               columnspan=grid_values[2] if len(grid_values) > 2 else 1)
+
+        self.height = rolling_offset + GraphNode.rows_per_graph
+
+    def update_title(self):
+        """
+        Allows you to update the name of the graph
+        """
+
+        if isinstance(self.items['title'], Entry):
+            name = self.items['title'].get()
+
+            self.items['title'].destroy()
+
+            self.items['title'] = Label(self.tab, text=name, font=("Arial Bold", 15), bg="#66AA33", borderwidth=1)
+
+            self.items['update_title']['text'] = "Change Name"
```

Comment:
Consider making an enum for controls. Also the text for the controls should be configurable constants.